README.md

```diff
@@ -174,19 +174,44 @@ After running the script, you will have 3 kind clusters created locally:
 
  If you want to create additional clusters, you can do so by running
  ```shell
- bash hack/add-cluster <cluster-name> <recreate>
+ bash hack/add-cluster <cluster-name> <recreate> <labels>
  ```
  This will spin up another kind cluster and register it against ArgoCD running in `kind-argocd-control-plane`
 
- #### Deploying a test appset
+ #### Deploying local test resources
 
-You can deploy a test appset to the default 3 clusters by running the following:
+You can deploy a test appset and a progressive sync object to your kind environment via:
 
 ```shell
-bash hack/deploy-test-appset.sh
+bash hack/redeploy-dev-resources.sh
 ```
 Feel free to extend the cluster generation section of the appset spec if you want to deploy it clusters that you have manually created.
 
+### Debugging
+
+```shell
+make debug
+```
+
+Invoking the command above should spin up a Delve debugger server in headless mode. You can then use your IDE specific functionality or the delve client itself to attach to the remote process and debug it.
+
+**NOTE**: On MacOSX, delve is currently unkillable in headless mode with `^C` or any other control signals that can be sent from the same terminal session. Instead, you'd need to run
+
+``` shell
+bash ./hack/kill-debug.sh
+``` 
+
+or
+
+``` shell
+make debug
+```
+
+from another terminal session to kill the debugger.
+
+### Debugging tests
+
+Delve can be used to debug tests as well. See `Test` launch configuration in `.vscode/launch.json`. Something similar should be achievable in your IDE of choice as well.
```

Comment:
As the rest of us use GoLand, could we add some instructions for it and mainly what we did yesterday. Essentially:

Update the make test step in the make file to be:
```
# Run tests
test: tools generate fmt vet manifests
	dlv test ./controllers
```

and set breakpoint inside the code using `runtime.Breakpoint()` and use the `dvl help` to see how the debugger CLI works.

Reply:
Those instructions are not Goland specific, i.e they would work with any editor. I can add them, but they are more of hacks than instructions. I'd rather spend sometime down the line for adding proper integration for Goland.

Reply:
Ok, I agree.